system_tests/bigquery.py

```diff
@@ -29,6 +29,11 @@
 
 DATASET_NAME = 'system_tests' + unique_resource_id()
 
+# We need to wait to stay within the rate limits.
+# The alternative outcome is a 403 Forbidden response from upstream.
+# See: https://cloud.google.com/bigquery/quota-policy
```

Comment:
Second question, shouldn't we add an `error_predicate` to make sure the `Forbidden` error is for rate-limiting and not for auth?

To build your predicate here is a [previous failure][1]

```python
Traceback (most recent call last):
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/system_tests/bigquery.py", line 87, in test_update_dataset
    dataset.create()
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/gcloud/bigquery/dataset.py", line 423, in create
    method='POST', path=path, data=self._build_resource())
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/gcloud/connection.py", line 347, in api_request
    error_info=method + ' ' + url)
gcloud.exceptions.Forbidden: 403 Exceeded rate limits: Your dataset exceeded quota for dataset.insert or dataset.update per table. For more information, see https://cloud.google.com/bigquery/troubleshooting-errors (POST https://www.googleapis.com/bigquery/v2/projects/precise-truck-742/datasets)
```

[1]: https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/149036776#L576